lib/api.js

```diff
@@ -104,7 +107,103 @@ module.exports = {
    * }
    * ```
    */
-  messages: []
+  // BUG previous messages aren't cleaned up
+  // messages: [],
+  /**
+   * Messages Error API
+   *
+   * push plugin errors to result.messages for reporting
+   *
+   * @memberof tree
+   *
+   * @method error
+   *
+   * @param  {String} plugin   Name ('posthtml-plugin')
+   * @param  {String} message  Error
+   *
+   * @return {Object} PostHTML Error Message
+   *
+   * ```js
+   * tree.error('posthtml-plugin', 'Error', node.location)
+   * ```
+   */
+  error: function (plugin, message, location) {
+    const err = new PluginError({
+      src: this.src,
```

Comment:
Maybe better as implemented as `location.src`

Reply:
indefinitely) in the case of `this.src`, it is clear what will be and the implication to the context of execution in the case of `location` is not entirely clear what to relate to `location`

Reply:
kk, yep `tree.source` or `this.source` etc would be perfectly fine